src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -285,8 +285,7 @@ Symbol* SystemDictionary::class_name_symbol(const char* name, Symbol* exception,
     return nullptr;
   }
   // Callers should ensure that the name is never an illegal UTF8 string.
-  assert(UTF8::is_legal_utf8((const unsigned char*)name,
-                             static_cast<int>(name_len), false),
+  assert(UTF8::is_legal_utf8((const unsigned char*)name, name_len, false),
```

Comment:
Is there where the > INT_MAX length got in from JNI?

Reply:
No, the failing path was in jniCheck.cpp. In this code we have already rejected strings > `Symbol::max_length`.